lib/rgeo/geos/interface.rb

```diff
@@ -163,15 +162,16 @@ def version
       #   never automatically generates a prepared geometry (unless you
       #   generate one explicitly using the <tt>prepare!</tt> method).
       def factory(opts = {})
-        if supported?
-          native_interface = opts[:native_interface] || Geos.preferred_native_interface
-          if opts[:has_z_coordinate] && opts[:has_m_coordinate]
-            ZMFactory.new(opts)
-          elsif native_interface == :ffi
-            FFIFactory.new(opts)
-          else
-            CAPIFactory.create(opts)
-          end
+        return unless supported?
```

Comment:
@keithdoggett here as well.

This makes me wonder if we want to make the move now of switching to raising as this will induce a lot more work checking all the places where we return nil. WDYT?

Reply:
I think that we probably should, but since this is a factory and not a geometry, it may not be as big of a deal. I guess we'd have to check for all places a factory is created then?